product_multi_relation/models/product_relation.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2013-2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# pylint: disable=api-one-deprecated
+"""Store relations (connections) between products."""
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductRelation(models.Model):
+    """Model product.relation is used to describe all links or relations
+    between products in the database.
+
+    This model is actually only used to store the data. The model
+    product.relation.all, based on a view that contains each record
+    two times, once for the normal relation, once for the inverse relation,
+    will be used to maintain the data.
+    """
+    _name = 'product.relation'
+    _description = 'Product relation'
+
+    left_product_id = fields.Many2one(
+        comodel_name='product.template',
+        string='Source Product',
+        required=True,
+        auto_join=True,
+        ondelete='cascade',
+    )
+    right_product_id = fields.Many2one(
+        comodel_name='product.template',
+        string='Destination Product',
+        required=True,
+        auto_join=True,
+        ondelete='cascade',
+    )
+    type_id = fields.Many2one(
+        comodel_name='product.relation.type',
+        string='Type',
+        required=True,
+        auto_join=True,
```

Comment:
Can you elaborate please? Not sure what you mean.

Reply:
It's not really common to use auto_join in all many2one relations (either Odoo would have done it per default in base ORM)

Reply:
These are intricacies that I yet have to master :-( However in product.py there is this referencing auto_join in search method:
```
        # because of auto_join, we have to do the active test by hand
        active_args = []
        if self.env.context.get('active_test', True):
            for arg in args:
                if (is_leaf(arg) and isinstance(arg[0], str) and
                        arg[0].startswith('search_relation')):
                    active_args = [('relation_all_ids.active', '=', True)]
                    break
```

If there is better way of doing this, please suggest.